test/dynamo/test_higher_order_ops.py

```diff
@@ -1631,31 +1637,26 @@ def f(x):
 
         self.assertEqual(result, x.sin() + x.cos())
         self.assertEqual(cnt.frame_count, 1)
+        self.assertEqual(len(counters["graph_break"]), 0)
         self.assertEqual(len(backend.graphs), 1)
         wrap_node = find_first_node(backend.graphs[0], wrap)
         self.assertTrue(len(wrap_node.args), 1)
         body_function = getattr(backend.graphs[0], wrap_node.args[0].name)
         self.assertEqual(op_count(body_function), 2)
 
-    def test_fallback_on_output_with_dict(self):
-        # We can likely support this in the future, I just don't want to deal
-        # with it right now
+    def test_output_with_dict(self):
         counters.clear()
         cnt = CompileCounter()
 
-        @torch.compile(backend=cnt)
+        @torch.compile(backend=cnt, fullgraph=True)
         def f(x):
             return wrap(lambda x: [{"a": -x}], x)
 
         x = torch.randn(3)
         result = f(x)
-        self.assertEqual(result, [{"a": -x}])
-        self.assertEqual(cnt.frame_count, 0)
-        assert_dict_matches_regex(
-            self,
-            dict(counters["graph_break"]),
-            {".*torch.* op returned non-Tensor dict call_function": 1},
-        )
+        self.assertEqual(result[0]["a"], -x)
+        self.assertEqual(cnt.frame_count, 1)
+        self.assertEqual(len(counters["graph_break"]), 0)
```

Comment:
Can we acquire the graph here and do a self.assertExpectedInlne here? We should see that the output of wrap was flattened